cardano-node/src/Cardano/Node/Tracing/Documentation.hs

```diff
@@ -90,12 +102,57 @@ import           Ouroboros.Network.TxSubmission.Outbound (TraceTxSubmissionOutbo
 
 
 
+data TraceDocumentationCmd
+  = TraceDocumentationCmd
+    { tdcConfigFile :: FilePath
+    , tdcOutput     :: FilePath
+    }
+
+parseTraceDocumentationCmd :: Opt.Parser TraceDocumentationCmd
+parseTraceDocumentationCmd =
+  Opt.subparser
+    (mconcat
+     [ Opt.commandGroup "Miscellaneous commands"
+     , Opt.metavar "trace-documentation"
+     , Opt.hidden
+     , Opt.command "trace-documentation" $
+       Opt.info
+         (TraceDocumentationCmd
+           <$> Opt.strOption
+               ( Opt.long "config"
+                 <> Opt.metavar "NODE-CONFIGURATION"
+                 <> Opt.help "Configuration file for the cardano-node"
+               )
+           <*> Opt.strOption
+               ( Opt.long "output-file"
+                 <> Opt.metavar "FILE"
+                 <> Opt.help "Generated documentation output file"
+               )
+           Opt.<**> Opt.helper)
+       $ mconcat [ Opt.progDesc "Generate the trace documentation" ]
+     ]
+    )
+
+deriving instance Generic UnversionedProtocol
+deriving instance Generic UnversionedProtocolData
+
+instance ToJSON UnversionedProtocol
+instance ToJSON UnversionedProtocolData
+
+runTraceDocumentationCmd
+  :: TraceDocumentationCmd
+  -> IO ()
+runTraceDocumentationCmd TraceDocumentationCmd{..} = do
+  docTracers
+    tdcConfigFile tdcOutput (Proxy @(CardanoBlock StandardCrypto))
+                            (Proxy @(NtN.ConnectionId LocalAddress))
+                            (Proxy @(NtN.ConnectionId NtN.RemoteAddress))
+
 -- Have to repeat the construction of the tracers here,
 -- as the tracers are behind old tracer interface after construction in mkDispatchTracers.
 -- Can be changed, when old tracers have gone
-docTracers :: forall blk t peer peerConn remotePeer resolverError ntnVersion ntnVersionData.
-  ( Show t
-  , forall result. Show (Query blk result)
+docTracers :: forall blk peer remotePeer.
+  (forall result. Show (Query blk result)
   , TraceConstraints blk
   , LogFormatting (ChainDB.InvalidBlockReason blk)
   , LedgerSupportsProtocol blk
```

Comment:
`LedgerSupportsProtocol blk` is unnecessary.

`RunNode blk` is pulling in 18 unnecessary constraints. I would consider creating a type synonym for the constraints needed.

Reply:
@Jimbo4350, I've removed what I could, and consolidated the constraints a bit.

..but removing the `RunNode` constraint is very difficult as of now, because of how tangled things are at the moment.
I've spent some hours on it, and it's just not easy at all.

It'll be far easier to deal with it at the point when we'll be removing the legacy system.

Reply:
Let me help you :smile: 
```
diff --git a/cardano-node/src/Cardano/Node/Tracing/Documentation.hs b/cardano-node/src/Cardano/Node/Tracing/Documentation.hs
index 887705e29..44d19cb3b 100644
--- a/cardano-node/src/Cardano/Node/Tracing/Documentation.hs
+++ b/cardano-node/src/Cardano/Node/Tracing/Documentation.hs
@@ -47,11 +47,15 @@ import           Cardano.Node.Handlers.Shutdown (ShutdownTrace)
 import           Cardano.Node.Startup
 import           Cardano.Node.TraceConstraints
 
+import           Ouroboros.Consensus.Block.Forging
 import           Ouroboros.Consensus.BlockchainTime.WallClock.Types (RelativeTime)
 import           Ouroboros.Consensus.BlockchainTime.WallClock.Util (TraceBlockchainTimeEvent (..))
 import           Ouroboros.Consensus.Cardano.Block
-import           Ouroboros.Consensus.Ledger.Query (Query)
-import           Ouroboros.Consensus.Ledger.SupportsMempool (ApplyTxErr, GenTxId)
+import           Ouroboros.Consensus.Ledger.Inspect
+import           Ouroboros.Consensus.Ledger.Query (Query, ShowQuery)
+import           Ouroboros.Consensus.Ledger.SupportsMempool (ApplyTxErr, GenTxId,
+                   LedgerSupportsMempool)
+import           Ouroboros.Consensus.Ledger.SupportsProtocol
 import           Ouroboros.Consensus.Mempool.API (TraceEventMempool (..))
 import           Ouroboros.Consensus.MiniProtocol.BlockFetch.Server
                    (TraceBlockFetchServerEvent (..))
@@ -59,6 +63,7 @@ import           Ouroboros.Consensus.MiniProtocol.ChainSync.Client (TraceChainSy
 import           Ouroboros.Consensus.MiniProtocol.ChainSync.Server (TraceChainSyncServerEvent)
 import           Ouroboros.Consensus.MiniProtocol.LocalTxSubmission.Server
                    (TraceLocalTxSubmissionServerEvent (..))
+import           Ouroboros.Consensus.Node.NetworkProtocolVersion
 import qualified Ouroboros.Consensus.Node.Run as Consensus
 import qualified Ouroboros.Consensus.Node.Tracers as Consensus
 import qualified Ouroboros.Consensus.Protocol.Ledger.HotKey as HotKey
@@ -85,8 +90,8 @@ import           Ouroboros.Network.PeerSelection.RootPeersDNS (TraceLocalRootPee
                    TracePublicRootPeers (..))
 import           Ouroboros.Network.Protocol.BlockFetch.Type (BlockFetch)
 import           Ouroboros.Network.Protocol.ChainSync.Type (ChainSync)
-import           Ouroboros.Network.Protocol.Handshake.Unversioned (UnversionedProtocolData (..),
-                   UnversionedProtocol (..))
+import           Ouroboros.Network.Protocol.Handshake.Unversioned (UnversionedProtocol (..),
+                   UnversionedProtocolData (..))
 import           Ouroboros.Network.Protocol.LocalStateQuery.Type (LocalStateQuery)
 import qualified Ouroboros.Network.Protocol.LocalTxSubmission.Type as LTS
 import           Ouroboros.Network.Protocol.TxSubmission.Type (TxSubmission)
@@ -152,11 +157,19 @@ runTraceDocumentationCmd TraceDocumentationCmd{..} = do
 -- Can be changed, when old tracers have gone
 docTracers :: forall blk peer remotePeer.
   ( TraceConstraints blk
+  , InspectLedger blk
+  , LedgerSupportsMempool blk
+  , LedgerSupportsProtocol blk
+  , Consensus.SerialiseNodeToNodeConstraints blk
   , LogFormatting peer
   , LogFormatting remotePeer
-  , Consensus.RunNode blk
+  , Show (BlockNodeToClientVersion blk)
+  , Show (BlockNodeToNodeVersion blk)
   , Show remotePeer
   , Show peer
+  , Show (ForgeStateUpdateError blk)
+  , Show (CannotForge blk)
+  , ShowQuery (BlockQuery blk)
   )
   => FilePath
   -> FilePath
```
What part in particular was difficult for you?

Reply:
@Jimbo4350, I've no idea how you came up with this list -- thank you a lot! : -)

Reply:
@Jimbo4350, applied your patch -- thanks a lot!